connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectResource.java

```diff
@@ -16,12 +16,15 @@
  */
 package org.apache.kafka.connect.runtime.rest.resources;
 
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import java.io.IOException;
 import java.util.concurrent.TimeUnit;
 
 /**
  * This interface defines shared logic for all Connect REST resources.
  */
-public interface ConnectResource {
+public interface ConnectResource extends ContainerRequestFilter {
```

Comment:
I found an alternative workaround, it's up to you if you like it better or worse than adding the no-op filter method.

The logic for emitting that warning in `Providers.checkProviderRuntime` is checking that there is something to constrain the component being checked to the current runtime (client vs server). The ContainerRequestFilter is marked as introducing a SERVER constraint in `Providers.getJaxRsProviderInterfaces`, and by extending it we are just inheriting that constraint.

Instead we can use the annotation mechanism intended for adding constraints to third-party SPIs to make the ConnectResource itself look like an SPI to Jersey. I tried adding this to the classes themselves next to the `Singleton` annotations, and it exploded. But it seems to work when the annotations are here on the interface instead.

```suggestion
@Contract
@ConstrainedTo(RuntimeType.SERVER)
public interface ConnectResource {
```

Reply:
They're both hacks, and both require implementing classes to be annotated with `@Singleton` to avoid unnecessary warnings, but this seems slightly cleaner (especially since it doesn't automatically implement the request filter interface that we may want to use for real later on).

I did some local experimenting and it seems like the `@ConstrainedTo` annotation isn't actually necessary--let me know if I'm missing something, though.

Reply:
> I did some local experimenting and it seems like the @ConstrainedTo annotation isn't actually necessary--let me know if I'm missing something, though.

I think you're right! I think that was from a slightly earlier iteration I had where i was trying to target [this first condition](https://github.com/eclipse-ee4j/jersey/blob/74bf3220a2dd5f2067e60c110d4981591e87604e/core-common/src/main/java/org/glassfish/jersey/internal/inject/Providers.java#L439) until I realized I couldn't get that one to work, and started targeting [this second condition](https://github.com/eclipse-ee4j/jersey/blob/74bf3220a2dd5f2067e60c110d4981591e87604e/core-common/src/main/java/org/glassfish/jersey/internal/inject/Providers.java#L443) instead.